source/features/default-to-rich-diff.tsx

```diff
@@ -0,0 +1,47 @@
+import select from 'select-dom';
+import features from '../libs/features';
+
+const childList = {childList: true};
+
+// To push new extensions if we want more file types to have rich diff as default
+const richDiffExtensions = ['svg'];
+
+const fileExtensionRegExp = new RegExp('\\.(' + richDiffExtensions.join('|') + ')$');
+
+const checkFileIsRichDiffable = (filename: string): boolean => fileExtensionRegExp.test(filename);
+
+const observer = new MutationObserver(([{addedNodes}]) => {
+	for (const node of addedNodes) {
+		const element = node as Element;
+		if (element.tagName === 'DIV') {
+			if (element.className === 'js-diff-progressive-container') {
+				observer.observe(element, childList);
+			} else {
+				setRichDiff(element);
+			}
+		}
+	}
+});
+
+function setRichDiff(node: Element) {
+	for (const fileHeader of select.all('.file-header', node)) {
+		if (checkFileIsRichDiffable(select('.file-info > a', fileHeader).title)) {
+			select('[aria-label="Display the rich diff"]', fileHeader).click();
```

Comment:
You can select this in the for of and click it. If file `isRichDiffable` then it will have this button

Reply:
Good point and I can gladly change this but do we want to default to the rich diff mode regardless of the type of file? I've added the check function to filter by the desired extensions (SVG in this case).

Reply:
Good question, we need to see what other files have rich diffs

Reply:
Maybe not, Markdown files also have a rich diff but their code is usually handcrafted. 

In this case I’d like to see the selector specify SVG files if the info is available in an attribute or className somewhere. If not, the `isRichDiffable` function should just replaced by `/\.svg$/i.test(filename)`. Adding extensions in the future is easy

Reply:
`.file-header` does have a `data-file-type` attribute with the extension of the file 👍 
I'll pick it from there.